dbt/include/trino/macros/materializations/incremental.sql

```diff
@@ -1,17 +1,71 @@
-{% macro dbt_trino_get_append_sql(tmp_relation, target_relation) %}
+{% macro dbt_trino_validate_get_incremental_strategy(raw_strategy) %}
+  {#-- Validate the incremental strategy #}
+
+  {% set invalid_strategy_msg -%}
+    Invalid incremental strategy provided: {{ raw_strategy }}
+    Expected one of: 'append', 'insert_overwrite'
+  {%- endset %}
+
+  {% if raw_strategy not in ['append', 'insert_overwrite'] %}
+    {% do exceptions.raise_compiler_error(invalid_strategy_msg) %}
+  {% endif %}
+
+  {% do return(raw_strategy) %}
+{% endmacro %}
+
+
+{% macro dbt_trino_get_insert_overwrite_sql(source_relation, target_relation, unique_key, dest_columns) %}
+
+    {% if unique_key %}
+    delete from {{ target_relation }}
+    where ({{ unique_key }}) in (
+        select ({{ unique_key }})
+        from {{ source_relation }}
+        );
+    {% endif %}
+
+    {{ dbt_trino_get_insert_into_sql(source_relation, target_relation, dest_columns) }}
+{% endmacro %}
+
+
+{% macro dbt_trino_get_insert_into_sql(source_relation, target_relation, dest_columns) %}
+
+    {%- set dest_cols_csv = get_quoted_csv(dest_columns | map(attribute='name')) -%}
 
-    {%- set dest_columns = adapter.get_columns_in_relation(target_relation) -%}
-    {%- set dest_cols_csv = dest_columns | map(attribute='quoted') | join(', ') -%}
     insert into {{ target_relation }}
-    select {{dest_cols_csv}} from {{ tmp_relation.include(database=true, schema=true) }};
+    select {{dest_cols_csv}} from {{ source_relation.include(database=false, schema=false) }}
 
-    drop table if exists {{ tmp_relation }};
```

Comment:
Perhaps that's the culprit of test failures.

Reply:
Right, these changes come from @findinpath in his feature/incremental branch which this was forked from. 

Should I re-do the changes forking from master instead?

Reply:
Try to add this line to your code and see if that helps.

Reply:
Yep, that did the trick. Thanks for your help @hovaesco